dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py

```diff
@@ -0,0 +1,753 @@
+):
+    # get occupations for which we have over 50 job adverts for
+    representative_occs = (
+        all_green_measures_df.groupby(occ_col)
+        .job_id.count()
+        .sort_values(ascending=False)
+        .where(lambda x: x >= min_num_job_ads)
+        .dropna()
+        .keys()
+        .tolist()
+    )
+
+    filtered_df = all_green_measures_df[
+        all_green_measures_df[occ_col].isin(representative_occs)
+    ].reset_index(drop=True)
+
+    print(f"Filtered from {len(all_green_measures_df)} rows to {len(filtered_df)}")
+
+    return filtered_df
+
+
+def create_green_skills_df(all_green_measures_df, occ_col="SOC_2020_name"):
+    all_green_measures_df["ENTS_GREEN_ENTS"] = all_green_measures_df.apply(
+        lambda x: x["ENTS"] + x["GREEN_ENTS"], axis=1
+    )
+
+    green_skills_df = (
+        all_green_measures_df[["job_id", occ_col, "ENTS_GREEN_ENTS"]]
+        .explode("ENTS_GREEN_ENTS")
+        .reset_index(drop=True)
+    )
+    green_skills_df["ENTS_GREEN_ENTS"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        merge_ents
+    )
+
+    green_skills_df["extracted_skill"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        lambda x: x[0] if isinstance(x, list) else None
+    )
+
+    green_skills_df["green_label"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        lambda x: x[1] if isinstance(x, list) and len(x) > 4 else "not_green"
+    )
+
+    green_skills_df["green_label_prob"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        lambda x: x[2] if isinstance(x, list) and len(x) > 4 else None
+    )
+
+    green_skills_df["skill_label"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        lambda x: x[3] if isinstance(x, list) and len(x) > 4 else None
+    )
+
+    green_skills_df["skill_id"] = green_skills_df["ENTS_GREEN_ENTS"].apply(
+        lambda x: x[4] if isinstance(x, list) and len(x) > 4 else None
+    )
+
+    green_skills_df = green_skills_df[green_skills_df["skill_label"] != ""]
+
+    # # # Remove the duplicate green skills per job advert
+    print(f"{len(green_skills_df)} rows deduplicated to ...")
+    green_skills_df.sort_values(by="extracted_skill", inplace=True)
+    green_skills_df.drop_duplicates(
+        subset=["job_id", "skill_label"], keep="first", inplace=True
+    )
+    green_skills_df = green_skills_df[~green_skills_df["extracted_skill"].isna()]
+    print(f"... {len(green_skills_df)} rows")
+
+    return green_skills_df
+
+
+def create_skill_df(skill_measures_df, job_id_col="job_id"):
+    full_skills_outputs = load_s3_data(
+        BUCKET_NAME,
+        f"outputs/data/green_skill_lists/20230914/full_esco_skill_mappings.json",
+    )
+
+    ents_explode = (
+        skill_measures_df[[job_id_col, "ENTS"]].explode("ENTS").reset_index(drop=True)
+    )
+    ents_explode["skill_label"] = ents_explode["ENTS"].apply(
+        lambda x: x[0] if x else []
+    )
+    ents_explode = ents_explode.explode("skill_label").reset_index(drop=True)
+
+    skill_match_thresh = 0.7
```

Comment:
could be nice to pull this out as a parameter if we want to play around with this later downstream

Reply:
yep good idea - I've added to a config